test/e2e/framework/pod/wait.go

```diff
@@ -268,8 +296,11 @@ func WaitForPodCondition(c clientset.Interface, ns, podName, conditionDesc strin
 			}
 			return true, err
 		} else if err != nil {
-			// TODO(#109732): stop polling and return the error in this case.
 			e2elog.Logf("Error evaluating pod condition %s: %v", conditionDesc, err)
+			var finalErr *FinalErr
+			if errors.As(err, &finalErr) {
+				return false, err
+			}
```

Comment:
if we want to expose it in the framework, is not better to have a function `IsFinal(err error) bool`

Reply:
Yes, that's a bit clearer. Added and pushed.